src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/__init__.py

```diff
@@ -3,16 +3,35 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
+from knack.events import EVENT_INVOKER_POST_PARSE_ARGS
+from knack.log import get_logger
 from azure.cli.core import AzCommandsLoader
-
+from azure.cli.core.commands import CliCommandType
 import azure.cli.command_modules.iot._help  # pylint: disable=unused-import
+from azure.cli.core.extension import extension_exists
+
+
+def handler(ctx, **kwargs):
+    cmd = kwargs.get('command', None)
+    if cmd and cmd.startswith('iot'):
+        if not extension_exists('azure_cli_iot_ext'):
```

Comment:
I think this should be `'azure-cli-iot-ext'`.

```
$ az extension list -otable
ExtensionType    Name               Version
---------------  -----------------  ---------
whl              azure-cli-iot-ext  0.4.1
$ python
>>> from azure.cli.core.extension import extension_exists
>>> extension_exists('azure_cli_iot_ext')
False
>>> extension_exists('azure-cli-iot-ext')
True
>>> 
```

Reply:
Good call. Thanks.